File: api/app/internal/model/enum/gender.go
package model

import (
	"fmt"
	"strconv"
)

type Gender int

const (
	GenderEmpty Gender = 0 + iota
	Male
	Female
	NoAnswer
)

// intからGender変換する処理
func ItoGender(i int) (Gender, error) {
	m := map[int]Gender{
		1: Male,
		2: Female,
		3: NoAnswer,
	}
	if m[i] == 0 {
		return GenderEmpty, fmt.Errorf("Genderではありません")
	}
	return m[i], nil
}

func (g *Gender) UnmarshalParam(src string) error {
	i, error := strconv.Atoi(src)
	if error != nil {
		return error
	}
	gender, error := ItoGender(i)
	if error != nil {
		return error
	}
	*g = gender
	return error
}

File: web-ui/src/models/enums/gender.ts
//export class Enum {

export enum Gender {
  Male = 1,
  Female = 2,
  NoAnswer = 3,
}

export function getGenderView(gender: Gender | null): string {
  if (gender == null) {
    return ""
  }
  const map: { [key: number]: string }  = {}
  // この書き方じゃないとGenderをkeyにできない
  map[Gender.Male] = "男性"
  map[Gender.Female] = "女性"
  map[Gender.NoAnswer] = "無回答"
  return map[gender] ?? ""
}

export function getGenderViewForDog(gender: Gender | null): string {
  if (gender == null) {
    return ""
  }
  const map: { [key: number]: string }  = {}
  // この書き方じゃないとGenderをkeyにできない
  map[Gender.Male] = "♂"
  map[Gender.Female] = "♀"
  map[Gender.NoAnswer] = "不明"
  return map[gender] ?? ""
}